src/ui/UI.java

```diff
@@ -69,6 +69,9 @@
 	private MenuControl menuBar;
 	private BrowserComponent browserComponent;
 	private RepositorySelector repoSelector;
+	
+	// Bview changes
+	private String pageContentOnLoad = "";
```

Comment:
Could this be kept track of in the browser component instead? That way it's concerned with its own state. The UI would call `browserComponent.hasPageChanged()` or something.